fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/Index.java

```diff
@@ -190,6 +190,25 @@ public Index(@Nonnull Index orig, @Nullable final IndexPredicate predicate) {
         this.lastModifiedVersion = orig.lastModifiedVersion;
     }
 
+    /**
+     * Copy constructor. This will create an index that is identical to the current <code>Index</code> with a given
+     * set of index options.
+     * @param orig original index to copy
+     * @param indexOptions the index options.
+     */
+    public Index(@Nonnull Index orig, @Nonnull final Map<String, String> indexOptions) {
```

Comment:
According to test gap analysis, this method hasn't been tested. Should it just be removed? I take it this was added to support setting the `efSearch` parameter via overriding index options, which I believe is no longer needed